lib/collection/src/shards/local_shard/clock_map.rs

```diff
@@ -204,66 +201,73 @@ impl RecoveryPoint {
-    pub fn has_any_lower(&self, other: &Self) -> bool {
-        self.clocks.iter().any(|(key, tick)| {
+    /// A clock that is present in this recovery point, but not in the `other`,
+    /// is always considered to be *newer*.
+    pub fn has_any_older_clocks_than(&self, other: &Self) -> bool {
+        self.clocks.iter().any(|(key, &tick)| {
             other
                 .clocks
                 .get(key)
-                .map_or(false, |other_tick| *tick < *other_tick)
+                .map_or(false, |&other_tick| tick < other_tick)
         })
     }
 
-    /// Extend this recovery point with new clocks from `clock_map`
+    /// Extend this recovery point with clocks that are only present in the `other`.
     ///
-    /// Clocks that we have not seen yet are added with a tick of `0`, because we must recover all
-    /// records for it.
+    /// Clocks that are not present in this recovery point are initialized to the tick `0`,
+    /// because we must recover all operations for them.
     ///
-    /// Clocks that we already have in this recovery point are not updated, regardless of their
-    /// tick value.
-    pub fn extend_with_missing_clocks(&mut self, other: &Self) {
+    /// Clocks that are already present in this recovery point are not updated.
+    pub fn initialize_clocks_missing_from(&mut self, other: &Self) {
         // Clocks known on our node, that are not in the recovery point, are unknown on the
         // recovering node. Add them here with tick 0, so that we include all records for it
-        for key in other.clocks.keys() {
-            self.clocks.entry(*key).or_insert(0);
+        for &key in other.clocks.keys() {
+            self.clocks.entry(key).or_insert(0);
         }
     }
 
-    /// Remove clocks from this recovery point, if they are equal in `other`
-    pub fn remove_equal_clocks(&mut self, other: &Self) {
-        for (key, tick) in &other.clocks {
-            if let Some(other_tick) = self.clocks.get(key) {
+    /// Remove clocks from this recovery point, that are equal to the clocks in the `other`.
+    pub fn remove_clocks_equal_to(&mut self, other: &Self) {
+        for (key, other_tick) in &other.clocks {
+            if let Some(tick) = self.clocks.get(key) {
                 if tick == other_tick {
                     self.clocks.remove(key);
                 }
             }
         }
     }
 
-    /// Remove a clock from this recovery point, if the given `clock_tag` describes an equal or
-    /// lower clock value.
-    pub fn remove_equal_or_lower(&mut self, clock_tag: ClockTag) {
-        let key = Key::from_tag(clock_tag);
-        if let Some(tick) = self.clocks.get(&key) {
-            if clock_tag.clock_tick <= *tick {
+    /// Remove a clock referenced by the clock tag from this recovery point, if the clock is
+    /// *newer or equal* than the tick in the tag.
+    pub fn remove_clock_if_newer_than_or_equal_to_tag(&mut self, tag: ClockTag) {
```

Comment:
I don't see how it is clear. The argument name does not change the fact, that you can read "if newer or equal" both ways. What are we comparing to what? `clock > tag` or `tag > clock`?

It does make more sense to read "left to right", e.g., `clock > tag`, but, like, do you *absolutely always* do the most logical thing, or are you, like, human? :D

In the end, you *have to* go read the documentation/code to verify that it is indeed `clock > tag`. Using `newer_than_or_equal_to_tag` clarifies `clock > tag`. It is clear to you, because you've been working with it for the last two weeks.

Reply:
`remove_ge_tag`?

The only reasonable assumption I can make about that is: we remove clocks, if greater or equal to the given tag.

Reply:
Or `retain_old_clocks(tag)` with a comment describing the exact condition.

Reply:
> `remove_ge_tag`?
> The only reasonable assumption I can make about that is: we remove clocks, if greater or equal to the given tag.

You still can read it both ways. Like, it literally reads as "remove greater or equal tag". Which is a lot like "remove if *tag* is greater of equal". Why are you so against the conjunctions? Just add `remove_ge_to_tag`, and it's clear. Though, I'd still prefer fully verbose one.

> Or `retain_old_clocks(tag)` with a comment describing the exact condition.

That's exactly what I *don't* want. I don't want to *have to* go read the comment. I want it to be obvious when you just read the method name. :/